### PR TITLE
Omit 'ref' from ButtonProps otherwise typing conflicts occur.

### LINKED ISSUE #10405: Button.d.ts seems incorrect. Error: "Types of property 'ref' are incompatible."
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.5.0

### Environment
OSX, Chrome

### Reproduction link
https://github.com/cdeutsch/antd-issue/
See reproduction below as well.

### Steps to reproduce
See the code and error in `MyButton.tsx` in reproduction.

### What is expected?
You should be able to spread all ButtonProps to a Button component.

### What is actually happening?
ButtonProps contains incorrect properties.

---
The bug was introduced in #10229  /cc @whtsky

### PR DESCRIPTION
**Pull request by cdeutsch**
Fixes #10405

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [x] Make sure that you add at least one unit test for the bug which you had fixed.
 
Can't add test because tests are in Javascript, and test would require Typescript.

### CODE REVIEW

**Comment by cdeutsch** [comment]
An alternative to this fix would be to revert #10229, and accept anything instead.

Right now you can't do something like:
```jsx
<Button data-test="mycomponent">Hello</Button>
```

```typescript
export interface ButtonProps {
    type?: ButtonType;
    htmlType?: string;
    icon?: string;
    shape?: ButtonShape;
    size?: ButtonSize;
    loading?: boolean | {
        delay?: number;
    };
    prefixCls?: string;
    className?: string;
    ghost?: boolean;

    // accept any:
    [elemName: string]: any;
}

```

File: components/button/button.tsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ export type ButtonType = 'default' | 'primary' | 'ghost' | 'dashed' | 'danger';
 export type ButtonShape = 'circle' | 'circle-outline';
 export type ButtonSize = 'small' | 'default' | 'large';
 
-export interface BaseButtonProps<T> extends Omit<React.HTMLProps<T>, 'size'> {
+export interface BaseButtonProps<T> extends Omit<Omit<React.HTMLProps<T>, 'ref'>, 'size'> {
```
Comment (yesmeck) [inline components/button/button.tsx]: `Omit<React.HTMLProps<T>, 'size' | 'ref'>`

**Comment by whtsky** [comment]
Ant Design's button currently does not forward native dom ref to the parent component ( https://reactjs.org/docs/refs-and-the-dom.html#exposing-dom-refs-to-parent-components ), so while this is a working fix, you'll still get the ref of antd's button component instead of actual dom.

**Comment by cdeutsch** [comment]
@whtsky it's not about forwarding the ref, it's about getting a ref to the React component (`<Button>`) I'm declaring. That's why the Ref's don't match.

File: components/button/button.tsx (hunk 1 above)
Comment (cdeutsch) [reply components/button/button.tsx]: Thanks @yesmeck. Fixed 👍

**Pull request merged**